File: frontend/src/types/session.ts
export interface Participant {
  id: number;
  session_id: string;
  role: "host" | "guest";
  channel: "local" | "remote";
  display_name: string;
  speaker_id: string | null;
  joined_at: string;
}

export interface Session {
  id: string;
  host_name: string;
  guest_url: string;
  source_lang: string;
  target_lang: string;
  glossary_id: string | null;
  diarize_enabled: boolean;
  status: string;
  created_at: string;
  ended_at: string | null;
  participants: Participant[];
}

export interface Utterance {
  channel: "local" | "remote";
  speaker_label?: string;
  language: string;
  original_text: string;
  translated_text: string;
  is_final: boolean;
  timestamp: number;
}

export interface GlossaryEntry {
  ja: string;
  vi: string;
  note: string | null;
}

File: backend/models/session.py
from dataclasses import dataclass

from pydantic import BaseModel


class SessionCreate(BaseModel):
    host_name: str
    source_lang: str = "ja"
    target_lang: str = "vi"
    diarize_enabled: bool = False


class ParticipantCreate(BaseModel):
    display_name: str


class ParticipantResponse(BaseModel):
    id: int
    session_id: str
    role: str
    channel: str
    display_name: str
    speaker_id: str | None = None
    joined_at: str


class SpeakerMappingResponse(BaseModel):
    id: int
    session_id: str
    channel: str
    dg_speaker_id: int
    display_name: str | None
    first_seen_at: float
    utterance_count: int


class SessionResponse(BaseModel):
    id: str
    host_name: str
    guest_url: str
    source_lang: str
    target_lang: str
    glossary_id: str | None = None
    diarize_enabled: bool = False
    status: str
    created_at: str
    ended_at: str | None = None
    participants: list[ParticipantResponse] = []


class UtteranceResponse(BaseModel):
    channel: str
    speaker_label: str | None = None
    language: str
    original_text: str
    translated_text: str
    is_final: bool
    timestamp: float


@dataclass
class TranscriptResult:
    channel: str
    detected_lang: str
    text: str
    is_final: bool
    speaker_id: int | None = None
